packages/http-client-csharp/emitter/src/type/input-examples.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { InputParameter } from "./input-parameter.js";
+import {
+  InputArrayType,
+  InputDictionaryType,
+  InputLiteralType,
+  InputModelType,
+  InputNullableType,
+  InputPrimitiveType,
+  InputType,
+  InputUnionType,
+} from "./input-type.js";
+import { OperationResponse } from "./operation-response.js";
+
+interface InputExampleBase {
+  kind: string;
+  name: string;
+  description: string;
+  filePath: string;
+  rawExample: any;
+}
+
+export interface InputHttpOperationExample extends InputExampleBase {
+  kind: "http";
+  parameters: InputParameterExampleValue[];
+  responses: Map<number, OperationResponseExample>;
+}
+
+export interface InputParameterExampleValue {
+  parameter: InputParameter;
+  value: InputTypeExampleValue;
+}
+
+export interface OperationResponseExample {
+  response: OperationResponse;
+  // headers: SdkHttpResponseHeaderExample[];
+  bodyValue?: InputTypeExampleValue;
+}
+
+export type InputTypeExampleValue =
+  | InputStringExampleValue
+  | InputNumberExampleValue
+  | InputBooleanExampleValue
+  | InputNullExampleValue
+  | InputAnyExampleValue
+  | InputArrayExampleValue
+  | InputDictionaryExampleValue
+  | InputUnionExampleValue
+  | InputModelExampleValue;
+
+export interface InputExampleTypeValueBase {
+  kind: string;
```

Comment:
`type:InputType` already contain the information of the `kind`. Do we need to duplicate this? otherwise, we need to validate if the `kind` is the same as the one of `type`

Reply:
this is duplicating the existing types from TCGC.
And it is marking the type of the "value", which might be different from the input type.
Taking datetime as an example